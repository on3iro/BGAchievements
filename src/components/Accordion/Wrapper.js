import styled from 'styled-components'

const Wrapper = styled.div`
  width: 100%;
  height: 90%;

  overflow: auto;

  div:last-child {
    margin-bottom: 0;
  }
`

export default Wrapper
