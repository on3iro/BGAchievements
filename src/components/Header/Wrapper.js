/**
  * Styled container for Header
  *
  *
  * @namespace Header.Wrapper
  * @memberOf Header
  */

import styled from 'styled-components'

const Wrapper = styled.header`
  position: fixed;
  top: 0px;
  margin-bottom: 0;

  min-height: 64px;
  height: 64px;
  width: 100%;

  background-color: ${props => props.theme.mainBG};
  color: ${props => props.theme.noneWhite};

  @media (max-width: 900px) {
    position: static;

    height: auto;
  }
`

export default Wrapper
