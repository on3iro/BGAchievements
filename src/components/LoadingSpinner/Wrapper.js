import styled from 'styled-components'

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  text-align: center;
`

export default Wrapper
