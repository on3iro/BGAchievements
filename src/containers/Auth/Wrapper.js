/**
  * Renders a styled div for the PasswordGenerator
  *
  * @namespace PasswordGenerator.Wrapper
  * @memberOf PasswordGenerator
  */

import styled from 'styled-components'

const Wrapper = styled.section`
  margin-top: 20px;
`

export default Wrapper
