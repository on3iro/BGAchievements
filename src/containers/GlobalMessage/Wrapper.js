import styled from 'styled-components'

const Wrapper = styled.div`
  margin: 80px 0 0 0;
  padding: 10px 20px;
  height: 40px;

  border-radius: 4px;

  background: ${props => props.error ? props.theme.warningHover : props.theme.primaryHover};
  opacity: ${props => props.empty ? '0' : '.6'};
  color: #fff;
`

export default Wrapper
