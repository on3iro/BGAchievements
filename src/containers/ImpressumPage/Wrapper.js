import styled from 'styled-components'

const Wrapper = styled.section`
  width: 100%;
  height: 100%;

  overflow: auto;
`

export default Wrapper
