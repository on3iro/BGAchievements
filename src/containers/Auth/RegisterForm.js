/*
  * The user registration form
  *
  * @namespace RegisterPage.RegisterForm
  * @memberOf Register
  */

import React from 'react'
import PropTypes from 'prop-types'
import { Field, reduxForm } from 'redux-form'

import H1 from 'components/H1'
import Button from 'components/Button'
import LoadingSpinner from 'components/LoadingSpinner'

import FormField from 'components/FormField'

export const validate = values => {
  const errors = {}
  const REQUIRED = 'Pflichtfeld'

  if (!values.email) {
    errors.email = REQUIRED
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Ungültige Email-Adresse'
  }

  if (!values.password) {
    errors.password = REQUIRED
  }

  if (!values.name) {
    errors.name = REQUIRED
  }

  if (!values.confirm) {
    errors.confirm = REQUIRED
  } else if (values.confirm !== values.password) {
    errors.confirm = 'Passwörter stimmen nicht überein!'
  }

  return errors
}

export const RegisterForm = props => {
  return (
    <form onSubmit={props.handleSubmit}>
      <H1>Registrierung</H1>
      <Field name='email' type='text'
        component={FormField} label='Email'
      />
      <Field name='password' type='password'
        component={FormField} label='Passwort'
      />
      <Field name='confirm' type='password'
        component={FormField} label='Passwort wiederholen'
      />
      {
        props.isLoading
          ? <Button onClick={(e) => e.preventDefault}>
            <LoadingSpinner controls />
          </Button>
          : <Button submit>Jetzt kostenlos registrieren</Button>
      }
    </form>
  )
}

RegisterForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
}

export default reduxForm({
  form: 'register',
  validate
})(RegisterForm)
