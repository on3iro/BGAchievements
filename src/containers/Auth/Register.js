/**
  * User registration page
  *
  * @namespace Register
  */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Redirect } from 'react-router-dom'

import * as selectors from './ducks/selectors'

import Button from 'components/Button'

import Wrapper from './Wrapper'
import RegisterForm from './RegisterForm'
import FormDivider from './FormDivider'

import { registerSubmit } from './ducks/actions'

export class Register extends Component {
  handleSubmit = values => {
    this.props.registerSubmit(values)
  }

  render () {
    const { from } = this.props.location.state || { from: { pathname: '/' } }

    return (
      this.props.hasRegistered || this.props.isLoggedIn
        ? (
          <Redirect to={from} />
        ) : (
          <Wrapper>
            <RegisterForm isLoading={this.props.isLoading} onSubmit={this.handleSubmit} />
            <FormDivider>Schon registriert?</FormDivider>
            <Button to='/login' secondary>Anmelden</Button>
          </Wrapper>
        )
    )
  }
}

Register.propTypes = {
  hasRegistered: PropTypes.bool,
  isLoading: PropTypes.bool,
  isLoggedIn: PropTypes.bool,
  location: PropTypes.object,
  registerSubmit: PropTypes.func.isRequired
}

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({
    registerSubmit
  }, dispatch)
}

export const mapStateToProps = (state) => {
  return {
    hasRegistered: selectors.getRegistered(state),
    isLoggedIn: selectors.getLoggedIn(state),
    isLoading: selectors.getLoadingState(state)
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Register)
