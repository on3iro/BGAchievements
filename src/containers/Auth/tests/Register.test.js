import React from 'react'
import { shallow } from 'enzyme'
import { Redirect } from 'react-router-dom'

import Wrapper from '../Wrapper'
import { Register, mapStateToProps, mapDispatchToProps } from '../Register'
import RegisterForm from '../RegisterForm'

const renderWrapper = newProps => {
  const props = {
    registerSubmit: jest.fn(),
    location: {
      state: 'test'
    },
    isLoggedIn: false,
    ...newProps
  }

  const renderedWrapper = shallow(<Register {...props} />)

  return {
    props,
    renderedWrapper
  }
}

describe('<Register />', () => {
  it('should render <Wrapper />', () => {
    const { renderedWrapper } = renderWrapper()
    expect(renderedWrapper.find(Wrapper).length).toMatchSnapshot()
  })

  it('should render <RegisterForm />', () => {
    const { renderedWrapper } = renderWrapper()
    expect(renderedWrapper.find(RegisterForm).length).toMatchSnapshot()
  })

  it('should render <Redirect /> if user is logged in', () => {
    const { renderedWrapper } = renderWrapper({ isLoggedIn: true })
    expect(renderedWrapper.find(Redirect).length).toMatchSnapshot()
  })

  it('should call registerSubmit() onSubmit', () => {
    const { renderedWrapper, props } = renderWrapper()

    renderedWrapper.instance().handleSubmit('test')

    expect(props.registerSubmit).toHaveBeenCalledWith('test')
  })

  describe('react-redux', () => {
    it('should map state to props', () => {
      expect(mapStateToProps({
        auth: {
          user: { hasRegistered: true, isLoggedIn: true },
          AuthRequestState: { isLoading: true }
        }
      })).toMatchSnapshot()
    })

    it('should map dispatch to props', () => {
      expect(mapDispatchToProps(() => null)).toMatchSnapshot()
    })
  })
})
