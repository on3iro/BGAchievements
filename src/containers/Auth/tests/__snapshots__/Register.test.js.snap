exports[`<Register /> react-redux should map dispatch to props 1`] = `
Object {
  "registerSubmit": [Function],
}
`

exports[`<Register /> react-redux should map state to props 1`] = `
Object {
  "hasRegistered": true,
  "isLoading": true,
  "isLoggedIn": true,
}
`

exports[`<Register /> should render <Redirect /> if user is logged in 1`] = `1`

exports[`<Register /> should render <RegisterForm /> 1`] = `1`

exports[`<Register /> should render <Wrapper /> 1`] = `1`
